src/entry.js

```diff
@@ -1,11 +1,13 @@
-const shared = require('./shared');
+const shared = require('./utils/shared');
 
 function StacheEntryPlugin() {
   const preload = (content, resourcePath, skyPagesConfig) => {
     try {
       const preloadPluginOrder = [
         require('./config'),
+        require('./json-data-element-attributes'),
         require('./include'),
+        require('./json-data-build-time'),
         require('./code-block'),
         require('./json-data'),
         require('./route-metadata'),
```

Comment:
1. `config` runs first to set up any config necessary that could affect later options.   At least that's the theory behind that one going first it seems to make sense.   
2. `element attributes`  this runs second, to replace any element attributes (such as `fileName`) with their appropriate values,   This needs to run before the `include` plugin, so it can load in the correct fileName when loading the snippet.
3. `include`,  this plugin has to run before we do any further content replacement, as the snippets that get pulled into the file from this include block, could very well contain stache jsonData themselves.
4. `build-time`,  Here we need to run over ALL the contents of the page before any further manipulation and replace all the `@buildtime` data objects,  this is required before we run the `code-block` plugin, so that we can use the `@buildtime` inside them
5. `code-block` needs to run before the `json-data` plugin, but after the `@buildtime` plugin, as here we want to escape the `{{` characters so they can render properly on the page without being interpolated by Angular and causing errors.  It has to run before the `json-data` plugin, otherwise it will try to add the elvis operator to the blocks where it likely isn't wanted.

`route-meta-data` doesn't need to be in a particular order as it is grabbing ALL HTML files and looping through them to create and set the meta data object.   
likewise I don't believe `template-reference-variable` is dependent on order either.